### ISSUE TITLE
Please add basic auth support

### ISSUE THREAD

**Issue by dennusb** [Graylog2/collector#16]
Because we are running Graylog at EC2 we've placed a proxy in front which is handling the Gelf requests, but first you need to authenticate with basic auth. Could you please add support for that in the collector?

**Comment by bernd** [comment]
You mean auth support for the collector registration requests against the Graylog server API on port 12900?

**Comment by dennusb** [comment]
Nope, basic auth for sending the Gelf messages on port 12201!

**Comment by dennisoelkers** [comment]
Are you using GELF over HTTP?

**Comment by dennisoelkers** [comment]
It might make sense to implement basic auth support for GELF over HTTP once we have support for GELF/HTTP in the gelfclient library.

**Comment by dennusb** [comment]
Yeah. We are using Gelf over HTTP, but to make it more secure there is a Squid in front processing authentication :)

**Comment by bernd** [comment]
I will file this as a feature request. We need the following for this:

1. HTTP support in gelfclient. (including basic auth) See: https://github.com/Graylog2/gelfclient/issues/9
1. Configuration support for HTTP GELF in the collector.
1. Basic auth support for the GELF HTTP input in Graylog server. See: https://github.com/Graylog2/graylog2-server/issues/335

**Comment by davidchua** [comment]
@dennusb how are you using squid to secure?
